src/application-manager.coffee

```diff
@@ -151,24 +153,11 @@ module.exports = class ApplicationManager extends EventEmitter
 						@images.cleanup()
 			createNetworkOrVolume: (step) =>
 				if step.model is 'network'
-					# TODO: These step targets should be the actual compose objects,
-					# rather than recreating them
-					Network.fromComposeObject({ @docker, @logger },
-						step.target.name,
-						step.appId,
-						step.target.config
-					).create()
+					@networks.create(step.target)
 				else
 					@volumes.create(step.target)
-			removeNetworkOrVolume: (step) =>
-				if step.model is 'network'
-					Network.fromComposeObject({ @docker, @logger },
-						step.current.name,
-						step.appId,
-						step.current.config
-					).remove()
-				else
-					@volumes.remove(step.current)
+			removeNetworkOrVolume: (step) ->
+				step.current.remove()
```

Comment:
imo it's a bit odd that models are created through their Manager but removed directly - I'd rather have everything go to the Manager for consistency, even if it doesn't add much at this point.

Reply:
+1 here

Reply:
Fair point, will change.